### PR TITLE
Restructure Repository Layout

### PR DESCRIPTION
**Pull request by jyhi**
This PR:

- Pulls `iota-client` out from `iota-client/`, making it a top-level crate
- Tweaks `Cargo.toml` to accommodate the change
- Tweaks `README.md`

... in the hope of making the package look more intuitive.

Bindings continue stay at `bindings/` as separate crates (are they?).

### CODE REVIEW

**Comment by jyhi** [comment]
There are failing CIs... too much damage

**Comment by jyhi** [comment]
It should be good now; the failing CIs need another PR to fix.

**Comment by Thoralf-M** [comment]
With this change `cargo run --example 10_mqtt --release` doesn't work anymore, because `mqtt` is not a default feature, could you change the comment in the 10_mqtt.rs example to `cargo run --example 10_mqtt --release --features="mqtt"`?

File: Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,84 @@
...
+zeroize = { version = "1.4.1", default-features = false }
+
+# sync
+ureq = { version = "2.2.0", default-features = false, features = ["tls", "json"], optional = true }
+# async
+reqwest = { version = "0.11.6", default-features = false, features = ["json", "rustls-tls", "blocking"], optional = true }
+futures = { version = "0.3.17", default-features = false, optional = true }
+
+# MQTT
+rumqttc = { version = "0.10", default-features = false, features = ["websocket"], optional = true }
```
Comment (Thoralf-M) [inline Cargo.toml]:
```suggestion
rumqttc = { version = "0.10.0", default-features = false, features = ["websocket"], optional = true }
```

File: Cargo.toml [hunk 2]
```diff
@@ -1,5 +1,84 @@
...
+version = "1.1.0"
+authors = [
+  "Thoralf Müller <thoralf.mueller@iota.org",
+  "Antony Dikov <antoni.dikov@iota.org>",
+  "Bing-Yang Lin <bingyang.lin@iota.org>",
+  "Lucas Nogueira <lucas.nogueira@iota.org>",
+  "Nathan Jaremko <nathan@jaremko.ca>",
+  "Yu-Wei Wu <wusyong9104@gmail.com>"
 ]
+edition = "2018"
```
Comment (Thoralf-M) [inline Cargo.toml:12]: Maybe we should also update to 2021, wdyt?

Comment (jyhi) [reply Cargo.toml:12]: _Do it! Just do it!_ (but in another PR of course)

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**